Merge ecflow workflow for running cycled parallel on WCOSS2

lgannoaa opened the pull request:
<!-- PLEASE READ -->
<!--
Before opening a PR, please note these guidelines:

- Each PR should only address ONE topic and have an associated issue
- No hardcoded or paths to personal directories should be present
- No temporary or backup files should be committed
- Any code that was disabled by being commented out should be removed
-->

**Description**
Management requested to merge the ecflow workflow in preparation to run parallel with the dev-v16 branch on WCOSS2 that includes end to end jobs. 
#684
<!-- Please include relevant motivation and context. -->
<!-- Please include a summary of the change and which issue is fixed. -->
<!-- List any dependencies that are required for this change. -->

<!-- Please provide reference to the issue this pull request is addressing. -->
<!-- For e.g. Fixes #IssueNumber -->

**Type of change**
- Include ecflow workflow files to run parallel on WCOSS2.
- Modify checkout and link script to checkout and install ecflow workflow for GFS cycled parallel on WCOSS2
- Fix config.gldas to allow uplevel assignment on CPCGAUGE
    Add WCOSS2 support to config.metp and config.vrfy
- Modified JGFS_ATMOS_CYCLONE_TRACKER to enable it for WCOSS2
    Add WCOSS2 support to verification and archive jobs arch.sh earc.sh metp.sh vrfy.sh
    Modify hpssarch_gen.sh to take log file from jobs
Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

**How Has This Been Tested?**
Currently is running cycled test on WCOSS2:
PSLOT: ecfops
HOMEgfs: /lfs/h2/emc/global/noscrub/Lin.Gan/para/packages/gfs.v16.2.0
COM: /lfs/h2/emc/ptmp/Lin.Gan/ecfops/com/prod/gfs/v16.2
log: /lfs/h2/emc/ptmp/Lin.Gan/ecfops/com/prod/gfs/v16.2/logs
EXPDIR: $HOMEgfs/parm/config
Verification Web site: https://www.emc.ncep.noaa.gov/gmb/Lin.Gan/metplus/ecfops
<!-- Please describe the tests that you ran to verify your changes and on the platforms these tests were conducted. -->
<!-- Provide instructions so we can reproduce. -->
<!-- Please also list any relevant details for your test configuration -->

<!-- Use the following as a guide to list your tests and delete options that are not relevant. Expand as necessary. -->
<!--

-->
  
**Checklist**

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] My changes generate no new warnings
- [ ] Any dependent changes have been merged and published

lgannoaa committed:
Modify checkout and link script to checkout and install ecflow workflow for GFS cycled parallel on WCOSS2

lgannoaa committed:
Fix config.gldas to allow uplevel assignment on CPCGAUGE
Add WCOSS2 support to config.metp and config.vrfy

lgannoaa committed:
Modified JGFS_ATMOS_CYCLONE_TRACKER to enable it for WCOSS2
Add WCOSS2 support to verification and archive jobs arch.sh earc.sh metp.sh vrfy.sh
Modify hpssarch_gen.sh to take log file from jobs

aerorahul commented on jobs/JGFS_ATMOS_CYCLONE_TRACKER:
```diff
@@ -138,6 +135,9 @@ elif [ $machine = LUNA -o $machine = SURGE -o $machine = WCOSS_C ] ; then # For
   mpirun -n 1 ${USHens_tracker}/extrkr_gfs.sh ${loopnum} ${cmodel} ${CDATE} ${pert} ${DATA}
+elif [ $machine = WCOSS2 ] ; then # For WCOSS2
+  machine=wcoss2
+  mpirun -n 1 ${USHens_tracker}/extrkr_gfs.sh ${loopnum} ${cmodel} ${CDATE} ${pert} ${DATA}
```
I think the correct usage on wcoss2 is `mpiexec` and not `mpirun`.  Please confirm.

WalterKolczynski-NOAA replied:
Really these should all be using the launcher variable set by the ${machine}.env file and not have all this machine-branching code here at all.

KateFriedman-NOAA replied:
Yep, should be `mpiexec` and I second Walter's comment, we can set this in `$machine.env`.

lgannoaa replied:
1. mpirun is ok to use because host is assigned. It is located in:
     /pe/intel/compilers_and_libraries_2020.4.304/linux/mpi/intel64/bin/mpirun
    Program run without issue see job log:
      /lfs/h2/emc/global/noscrub/Lin.Gan/git/feature-ops-wcoss2/jobs/gfs_emc_vrfy_00.o5065799
2. Using launcher variable in jjob is not within the scope of this work. This job exist in workflow not a new job in wcoss2. Looking at the env file, I do not see every job within workflow are using it. Recommend to handle this request in a separate GitHub issue.

aerorahul replied:
1.  Per Carolyn Pasti's presentation on WCOSS2, the correct usage is `mpiexec` and not `mpirun`.
Please see the example from slide 53 from [this](https://www2.emc.ncep.noaa.gov/forms/training/WCOSS2gettingStarted.pdf) presentation.
2. Use of `launcher` is out of the scope of this work.  An issue can be created to resolve this.

lgannoaa replied:
I do not see problem of using mpirun. The code manager @JiayiPeng-NOAA  should be contacted, if request for change. He is ok with it and checked the output found no issue.

aerorahul replied:
Please work w/ whoever it is to bring up to the correct and acceptable usage.  `mpirun` is unacceptable per WCOSS2 instructions.

arunchawla-NOAA replied:
Please use mpiexec, all wcoss2 codes are using that. We have had extensive discussions with NCO on this already

lgannoaa replied:
Removed the use of mpirun.

aerorahul commented on jobs/JGLOBAL_WAVE_PREP:
```diff
@@ -66,20 +66,20 @@ export COMIN=${COMIN:-${ROTDIR}/${CDUMP}.${PDY}/${cyc}/$COMPONENT}
-  export COMIN_WAV_ICE=${COMIN_WAV_ICE:-$ROTDIR/$RUN.$PDY/$cyc/atmos}
-  export COMIN_WAV_RTOFS=${COMIN_WAV_RTOFS:-$ROTDIR}
-fi
+export COMIN_WAV_ICE=${COMIN_WAV_ICE:-$(compath.py ${envir}/obsproc/${obsproc_ver})}/${CDUMP}.${PDY}/${cyc}/atmos
+export COMIN_WAV_RTOFS=${COMIN_WAV_RTOFS:-$(compath.py ${envir}/${WAVECUR_DID}/${rtofs_ver})}
```
The code on the left is for execution when NCO runs this.
Is this no longer NCO specific?
What about on R&D machines?

Does using the two lines above from 69, 70 remove the need for the commented code below on R&D machines?
Please confirm.

KateFriedman-NOAA replied:
@JessicaMeixner-NOAA Should review this change, this is significant and needs to be vetted for use in ops since it will end up in the v16.3 upgrade. We went back and forth on this script a lot so changes must be considered/tested very carefully. Also, why is a block commented out but left in (lines 72-82)?

lgannoaa replied:
@JessicaMeixner-NOAA is in control of this JJob, I will take recommendation from her. The reason for this change is the original logic unconditionally make slink to file using global dump directory structure without first checking the availability of the source file.

aerorahul replied:
@JessicaMeixner-NOAA is unavailable and otherwise occupied with other more pressing issues.
Please highlight specifically what the issue is and how it is impacting WCOSS2 while keeping in mind that this also should work on other machines.
If there is a need for another conditional, please illustrate the need.
If there is a failure encountered with the current logic, please indicate it in a gist for e.g.

JessicaMeixner-NOAA replied:
The availability of the source file is handled in the script, for example If the current day's RTOFS is not available, the script will use the RTOFS from the day before: 
https://github.com/NOAA-EMC/global-workflow/blob/operations/scripts/exgfs_wave_prep.sh#L647-L674 

If that RTOFS file is not there, WW3 prep will error out: 
https://github.com/NOAA-EMC/global-workflow/blob/operations/scripts/exgfs_wave_prep.sh#L711-L721 

I share in the concern expressed by @aerorahul  that the changes you made here will break this for R&D machines and I'm concerned this would also not work on WCOSS2 for EMC or NCO without the commented out sections of the code.

arunchawla-NOAA replied:
please revert this change. It is already working on wcoss2 and has been extensively tested

lgannoaa replied:
Reverted as requested.

aerorahul commented on jobs/rocoto/arch.sh:
```diff
@@ -12,12 +12,6 @@
 
-###############################################################
-# Source FV3GFS workflow modules
-. $HOMEgfs/ush/load_fv3gfs_modules.sh
```
This will be required on the R&D machines to load HPSS module, no?
Can you provide a reason why this is deleted?

KateFriedman-NOAA replied:
I second Rahul's question, why is this deleted?

lgannoaa replied:
In wcoss2 source this load_fv3gfs_modules.sh script break things. Please check with changing log, NCO Wei has removed load_fv3gfs_modules.sh on his solution too. If this is needed, maybe it can be modified as machine dependent.

aerorahul replied:
NCO does not use this script in any form whatsoever.  Please correct me if I am incorrect.

arunchawla-NOAA replied:
why are we changing rocoto scripts in this PR? are they not working for rocoto ?

lgannoaa replied:
NCO removed scripts source $HOMEgfs/ush/load_fv3gfs_modules.sh due to environment variable been overwrite to default that break ecflow workflow.

aerorahul commented on jobs/rocoto/earc.sh:
```diff
@@ -13,12 +13,6 @@
 
-###############################################################
-# Source FV3GFS workflow modules
-. $HOMEgfs/ush/load_fv3gfs_modules.sh
```
same question as above.

lgannoaa replied:
Same asnwer as before

aerorahul commented on jobs/rocoto/metp.sh:
```diff
@@ -13,14 +13,6 @@
-###############################################################
-echo
-echo "=============== START TO SOURCE FV3GFS WORKFLOW MODULES ==============="
-. $HOMEgfs/ush/load_fv3gfs_modules.sh
```
This might be non-consequential to remove, but good to verify on a R&D machine.

WalterKolczynski-NOAA replied:
I know without even trying that MET+ uses modules. There are specific modules just for that in module_base.

lgannoaa replied:
The load_fv3gfs_modules.sh conversation is happened in above. It is a repeat conversation here. We can continue this topic in the original message dialog above. Will mark as resolved here.

aerorahul commented on jobs/rocoto/vrfy.sh:
```diff
@@ -88,10 +80,16 @@ if [ $VRFYFITS = "YES" -a $CDUMP = $CDFNL -a $CDATE != $SDATE ]; then
-
     export RUN_ENVIR_SAVE=$RUN_ENVIR
     export RUN_ENVIR=$OUTPUT_FILE
+    export COMDAY=$PBS_O_WORKDIR
```
$PBS_O_WORKDIR is a `PBS` specific variable and not defined on R&D machines w/ Slurm.
Please verify its compatibility on R&D machine.

WalterKolczynski-NOAA replied:
Should this not be $DATA?

lgannoaa replied:
Yes. The $PBS_O_WORKDIR is for PBS. Only pre implementation parallel are running the EMC verification jobs. I will revert this change.

aerorahul commented on parm/config/config.vrfy:
```diff
@@ -16,26 +16,24 @@ export CDFNL="gdas"               # Scores verification against GDAS/GFS analysi
-export VRFYMINMON="YES"           # GSI minimization monitoring
+export VSDB_STEP1="NO"            # Disable VSDB
+export VSDB_STEP2="NO"            # Disable VSDB
+export VRFYG2OBS="NO"             # Grid to observations, see note below if turning ON
+export VRFYPRCP="NO"              # Precip threat scores
+export VRFYRAD=${VRFYRAD:-"YES"}  # Radiance data assimilation monitoring
+export VRFYOZN=${VRFYOZN:-"YES"}  # Ozone data assimilation monitoring
+export VRFYMINMON=${VRFYMINMON:-"YES"} # GSI minimization monitoring
 export VRFYTRAK="YES"             # Hurricane track verification
 export VRFYGENESIS="YES"          # Cyclone genesis verification
-export VRFYFSU="NO"               # Cyclone genesis verification (FSU)
-export RUNMOS="NO"                # whether to run entire MOS package
+export VRFYFSU="YES"               # Cyclone genesis verification (FSU)
+export RUNMOS="YES"                # whether to run entire MOS package
```
Can you please explain why the defaults are changed?
Is it intended that `dev_v16` on R&D machines will run with these new defaults?

lgannoaa replied:
Tracker code manager Jiayi requested the VRFYFSU be turn on.
Will reverse this change to the original.

aerorahul replied:
Ok. And what about he other defaults; e.g. 
```
VSDB_STEP1
VRFYG2OBS
VRFYPRCP
RUNMOS
```
Is there a need for providing external over-rides for these?
```
VRFYRAD
VRFYOZN
VRFYMINMON

```

lgannoaa replied:
The new ecflow run verification job VRFYRAD, VRFYOZN, and VRFYMINMON differently. I adopt the same solution here. Doing so is to test and ensure job run as designed.
I will revert it.

aerorahul commented on parm/config/config.vrfy:
```diff
@@ -16,26 +16,24 @@ export CDFNL="gdas"               # Scores verification against GDAS/GFS analysi
-
-    export fit_ver="newm.1.3"
-    export fitdir="$BASE_GIT/verif/global/Fit2Obs/${fit_ver}/batrun"
+    export fitdir="/lfs/h2/emc/global/noscrub/emc.global/git/Fit2Obs"
```
This is a hard-wired path to WCOSS2.  Is this appropriate or a temporary measure?

aerorahul replied:
Seems it appropriate place would be in the WCOSS2 if-block, if necessary.

KateFriedman-NOAA replied:
This should not be a hardcoded path...it should be `$BASE_GIT/Fit2Obs`. What happened to the `${fit_ver}/batrun` part of the path, is it no longer needed?

lgannoaa replied:
It is just temporary. Will fill in when group account installed it. It will not be hardcoded path at the end.

aerorahul replied:
The directory `/lfs/h2/emc/global/noscrub/emc.global/git/Fit2Obs` does not exist.

![Screen Shot 2022-03-24 at 11 47 07 AM](https://user-images.githubusercontent.com/11394126/159955821-15c97864-ee6f-4443-a38e-a9c7568091f2.png)

`VRFYFITS` requires this utility to be installed prior to testing.  
Is it possible that this was built, installed, and tested from your personal space?
In that case, could you please include instructions to build and install to @KateFriedman-NOAA so she can install it in the `save/emc.global/git` space similar to the obsproc packages in the above screen shot.

The same question applies to `ENS_TRACKER`.  Please provide instructions to build and install that utility that typically goes under `HOMEens_tracker`.

lgannoaa replied:
The path in question here is only a temp place holder.

aerorahul commented on parm/config/config.vrfy:
```diff
@@ -142,22 +141,25 @@ fi
 fi
-
+if [ $machine = "WCOSS2" ] ; then
+    export HOMEens_tracker=/lfs/h2/emc/global/noscrub/emc.global/git/ens_tracker.${ens_tracker_ver}
```
I believe @KateFriedman-NOAA will soon have an installation of the tracker in `BASE_GIT`.  Is there a reason to not have `$BASE_GIT` instead of the hard-wired path?

KateFriedman-NOAA replied:
We should be trying to consolidate this whole section to just be 
`export HOMEens_tracker=$BASE_GIT/ens_tracker.${ens_tracker_ver}`
...where `ens_tracker_ver` can be set in `run.ver`. If the tracker version were to differ between machines then `ens_tracker_ver` would go into each `$target.ver` file.

lgannoaa replied:
Similar to the comment above. This is temporary.

aerorahul commented on sorc/checkout.sh:
```diff
@@ -99,4 +99,13 @@ else
     echo 'Skip. Directory verif-global.fd already exist.'
 fi
 
+echo wcoss2_ecf_cycled checkout ...
+if [[ ! -d wcoss2_ecf_cycled ]] ; then
+    rm -f ${topdir}/checkout-wcoss2_ecf_cycled.log
+    git clone --branch feature/wcoss2_ecf_cycled https://github.com/lgannoaa/ecflow.git wcoss2_ecf_cycled >> ${topdir}/checkout-wcoss2_ecf_cycled.log 2>&1
+    cd ${topdir}
+else
+    echo 'Skip. Directory wcoss2_ecf_cycled already exist.'
+fi
```
What is this and why are these scripts to run ecflow not in the global-workflow repository?

More importantly, this is a private repository!!!

![Screen Shot 2022-03-23 at 5 39 08 PM](https://user-images.githubusercontent.com/11394126/159800276-e94ab8e5-ff95-4f67-82a9-830bf02046ed.png)

KateFriedman-NOAA replied:
Strongly second @aerorahul 's question...what is this?

lgannoaa replied:
Checking out package and link is standard practice in global-workflow. We should do the same for ecf directory. Maybe we can create an official repo for ecflow?

aerorahul replied:
Storing files in a Private repository is not standard practice anywhere in NOAA-EMC.
Has a need been shown for having a separate set of scripts;  let alone in a users personal account? 
Adding a repository with ecf files is outside of the scope of this PR that enables cycled parallel on WCOSS2 with the verification components.

lgannoaa replied:
Never said storing file in private repository is a standard practice. Hence the suggestion to create a official repo for ecflow workflow files.
As requested, I will modify this approach to overwrite ecf directory under $HOMEgfs to the new version.

aerorahul replied:
Yes. Please open a separate PR with those changes related internal to `ecf/`
I would like to see the changes in context of NCO's use of them.

arunchawla-NOAA replied:
I want to minimize the number of repos that are used to check out a full application from, this is making the global workflow unsustainable. A repo should only really be created when it can stand on its own and be run. Otherwise it should be in one repository only

lgannoaa replied:
@aerorahul This PR will be the PR for ecflow. I do not understand why it need to have another new PR.

lgannoaa replied:
@arunchawla-NOAA Thank you for your guidance.

aerorahul commented on sorc/link_fv3gfs.sh:
```diff
@@ -339,6 +339,15 @@ else
+#--link ecflow directories
+#------------------------------
+
+if [ $machine == "wcoss2" ]; then
+  cd $pwd/..
+  [[ -s ecf ]] && mv ecf ecf-old
+  $SLINK $pwd/wcoss2_ecf_cycled/ecf
+fi
```
In light of the above comment, this is not a valid solution.

KateFriedman-NOAA replied:
Why not commit ecf file changes into the g-w repo? What is this?

lgannoaa replied:
This is the same question, please communicate in the same communication dialog box above.

aerorahul replied:
It is not the same.  
The comment above is a question about introducing a new repository and its associated issues.
Comment here is about moving/linking/duplicating existing scripts and then using them.

And comments are made where they are relevant, even if related.

lgannoaa replied:
Agreed to modify the solution above. This is part of the same solution that will be modified.

aerorahul commented on ush/hpssarch_gen.sh:
```diff
@@ -234,12 +234,13 @@ if [ $type = "gdas" ]; then
-   fi
-  done
-  echo  "./logs/${CDATE}/gdaspost*.log               " >>gdas.txt
+#  for fstep in prep anal gldas fcst vrfy radmon minmon oznmon; do
+#   if [ -s $ROTDIR/logs/${CDATE}/gdas${fstep}.log ]; then
+#     echo  "./logs/${CDATE}/gdas${fstep}.log         " >>gdas.txt
+#   fi
+#  done
+#  echo  "./logs/${CDATE}/gdaspost*.log               " >>gdas.txt
```
can you please explain what is the intent behind this?
Are logs not supposed to go to archive?

lgannoaa replied:
This script wired in the log file using rocoto workflow format. It is modified to include all log file in the log directory, so ecflow log file can also be included in backup set.

aerorahul replied:
Does `ecflow` run in development mode for all users?
Does `ecflow` jobs archival jobs call `arch.sh` or `earc.sh`? 
I have not seen those jobs in the `ecf/`.
So how are these being accessed/called?

lgannoaa replied:
User does not necessary to run ecflow in development mode.
I developed a solution to enable it EMC vrfy and archive jobs. As communicated above, the new ecf directory will replace the original ecf directory in next commit. We can talk about the solution as needed.

arunchawla-NOAA replied:
perhaps I am missing something but just looking at the logic of the script we will no longer be archiving gdasprep.log, gdasanal.log etc ? Looks like we are only archiving gdaspost*.log ? Or perhaps I am missing the bigger picture ?

lgannoaa replied:
Line 237-242 hardwired to only backup rocoto workflow log file. ecflow workflow does not use the same log file name. Therefore, removed hardwired name and use the line 243 and 378 to take in all log file found in log directory.

aerorahul commented on ush/hpssarch_gen.sh:
```diff
@@ -371,10 +372,10 @@ if [ $type = "enkfgdas" -o $type = "enkfgfs" ]; then
-   echo  "logs/${CDATE}/${CDUMP}${fstep}*.log        " >>enkf${CDUMP}.txt
-  done
-
+#  for fstep in eobs eomg ecen esfc eupd efcs epos ; do
+#   echo  "logs/${CDATE}/${CDUMP}${fstep}*.log        " >>enkf${CDUMP}.txt
+#  done
+  echo  "logs/${CDATE}/enkfgdas*.log        " >>enkf${CDUMP}.txt
```
same question as above for `arch.sh`.

lgannoaa replied:
Same solution logic as explained above.

aerorahul replied:
Same follow-up question as above.

lgannoaa replied:
ok

aerorahul commented on parm/config/config.metp:
```diff
@@ -14,56 +14,86 @@ export RUN_PRECIP_STEP1="YES"    # Run precip verification using METplus
-    fi
-fi
+## EMC_VERIF_GLOBAL SETTINGS
+export HOMEverif_global=${HOMEgfs}/sorc/verif-global.fd
```
I believe we might need SME to ensure these are correct for the `gdas` cycle, since the `gfs` cycle if-block has been removed.

I also note additional changes are made which might need checking due to the change in defaults in `config.vrfy`.

KateFriedman-NOAA replied:
@malloryprow should review this change since she built this config and updates it for us usually

malloryprow replied:
@KateFriedman-NOAA With the way things are set up now, the change is fine, as the metp jobs only run for the "gfs" cycle and these settings are for the gfsmetp jobs. However, in the future if the need ever arises that some sort of METplus job needs to be setup and configured to run in the "gdas" cycle, then the if statements would need to go back.

KateFriedman-NOAA replied:
Noted, thanks @malloryprow !

aerorahul reviewed (commented):
@lgannoaa 
Thank you for your PR and changes.
I have a few comments that need clarifications.

While I understand this branch is currently in preparation for being used on the R&D machines, some of the edited/removed code might need to be added back.  To that end, we might need a different approach.

We can discuss that at a later time as the review progresses.

WalterKolczynski-NOAA reviewed (changes requested):
I have pretty much the same questions as Rahul. Also, if code _is_ to be removed, it should just be removed, not commented out.

KateFriedman-NOAA commented on jobs/rocoto/vrfy.sh:
```diff
@@ -12,14 +12,6 @@
 ## cyc    : current cycle (HH)
 ###############################################################
 
-###############################################################
-echo
-echo "=============== START TO SOURCE FV3GFS WORKFLOW MODULES ==============="
-. $HOMEgfs/ush/load_fv3gfs_modules.sh
-status=$?
-[[ $status -ne 0 ]] && exit $status
-
-
```
Why is this removed?

lgannoaa replied:
Duplicated comment.

KateFriedman-NOAA commented on parm/config/config.gldas:
```diff
@@ -10,7 +10,11 @@ echo "BEGIN: config.gldas"
 export GLDASSH=$HOMEgfs/scripts/exgdas_atmos_gldas.sh
 export gldas_spinup_hours=72
-export CPCGAUGE=$DMPDIR
+if [ $RUN_ENVIR = "emc" ]; then
+  export CPCGAUGE=${DMPDIR:-/lfs/h2/emc/global/noscrub/emc.global/dump}
+else
+  export CPCGAUGE=""
+fi
```
Please explain this change. What is this fixing? What happens without this and what happens with this?

lgannoaa replied:
Revert the change.

KateFriedman-NOAA commented on parm/config/config.vrfy:
```diff
@@ -52,6 +50,9 @@ if [ $VRFYFITS = "YES" ]; then
     elif [ $machine = "WCOSS_DELL_P3" ]; then
         export PREPQFITSH="$fitdir/subfits_dell_nems"
+    elif [ $machine = "WCOSS2" ]; then
+        export FITDIR=$fitdir
```
Why are there two fitdir paths? FITDIR and fitdir? Why not just use one?

lgannoaa replied:
If take a look at this file history, it will show this 2 fitdir solution has been used before. Just adopting same solution.

KateFriedman-NOAA commented on parm/config/config.vrfy:
```diff
@@ -101,8 +102,6 @@ fi
 
 if [ $VRFYRAD = "YES" -o $VRFYMINMON = "YES" -o $VRFYOZN = "YES" ]; then
 
-    export envir="para"
```
Is this no longer needed?

lgannoaa replied:
This is hardwired. It is removed to allow upper level assignment.

KateFriedman-NOAA commented on parm/config/config.vrfy:
```diff
@@ -142,22 +141,25 @@ fi
 #-------------------------------------------------
 
-export ens_tracker_ver=v1.1.15.1
+export ens_tracker_ver=v1.1.15.5
```
This should be set in a version file now (e.g. `versions/run.ver`).

lgannoaa replied:
Used the same approach as the original. If it needs to be modified, recommend to do in a separate issue ticket.

aerorahul replied:
When we ported to WCOSS2, as @KateFriedman-NOAA indicates, this needs to be set in the version file and not in the current location.
If unable to do this in this PR, then please open an issue.  It will be assigned appropriately.
It is in the scope of this work.

KateFriedman-NOAA reviewed (changes requested):
I second all of the other developer questions/comments. We need to understand why most of these changes are happening. I see zero incoming changes to ecf files, which is what I expected to see in this PR.

lgannoaa commented:
> I have pretty much the same questions as Rahul. Also, if code _is_ to be removed, it should just be removed, not commented out.

Historically, many workflow scripts have comments. I prefer to not remove those comments in this scope of work. Maybe  address as an separate issue as needed.

aerorahul reviewed (commented):
Please see comments.

lgannoaa committed:
Replace ecf directory with cycled gfs ecflow workflow files
Add missing file JGDAS_ATMOS_CYCLONE_TRACKER
Modified:   jobs/rocoto/vrfy.sh with default log location
modified:   parm/config/config.gldas remove DMPDIR overwrite
modified:   parm/config/config.vrfy using original default

lgannoaa committed:
Revert JGLOBAL_WAVE_PREP to original

lgannoaa commented:
Close this PR as requested
